kolibri/core/auth/api.py

```diff
@@ -786,6 +786,27 @@ class MembershipViewSet(BulkDeleteMixin, BulkCreateMixin, viewsets.ModelViewSet)
     serializer_class = MembershipSerializer
     filterset_class = MembershipFilter
 
+    def create(self, request):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        self.perform_create(serializer)
+
+        response_data = {"created": serializer.data}
+        status_code = status.HTTP_201_CREATED
+
+        if hasattr(serializer, "invalid_items") and serializer.invalid_items:
+            status_code = status.HTTP_207_MULTI_STATUS
+            response_data["invalid"] = [
```

Comment:
(Nitpick) Initially, it was not evident to me why an `invalid_items` property would be set if the `is_valid` method is setting `raise_exception=True` and in theory it would raise an exception if the data is invalid, but this is true for bulk creations where some items may be invalid but not all of them, and therefore it wouldn't raise the exception. It would be great if we could have some comments around here explaining this.